spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -176,4 +182,13 @@ private void validatePartitionTransforms(PartitionSpec spec) {
           String.format("Cannot write using unsupported transforms: %s", unsupported));
     }
   }
+
+  private boolean checkNullability(DataSourceOptions options) {
+    boolean sparkCheckNullability = Boolean.parseBoolean(lazySpark.conf()
+        .get("spark.sql.iceberg.check-nullability",
+            Boolean.toString(TableProperties.CHECK_NULLABILITY_DEFAULT)));
```

Comment:
Can this be defaulted here to true? This shouldn't be in `TableProperties`.

Reply:
Ok. Will use concrete boolean value.